napari/_qt/qt_color_dialog.py

```diff
@@ -0,0 +1,257 @@
+
+        Parameters
+        ----------
+        parent : QWidget, optional
+            parent widget, by default None
+        initial_color : ColorType, optional
+            Starting color, by default None
+        tooltip : str, optional
+            Tooltip when hovering on the swatch,
+            by default 'click to set color'
+        """
+        super().__init__(parent=parent)
+        self.setObjectName('QColorSwatchEdit')
+
+        layout = QHBoxLayout()
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.setSpacing(6)
+        self.setLayout(layout)
+
+        self.lineEdit = QColorLineEdit(self)
+        self.lineEdit.editingFinished.connect(self._on_lineedit_edited)
+
+        self.colorSwatch = QColorSwatch(self, tooltip=tooltip)
+        self.colorSwatch.color_changed.connect(self._on_swatch_changed)
+        self.setColor = self.colorSwatch.setColor
+        if initial_color is not None:
+            self.setColor(initial_color)
+
+        layout.addWidget(self.colorSwatch)
+        layout.addWidget(self.lineEdit)
+
+    def color(self):
+        """Return the current color."""
+        return self.colorSwatch.color()
+
+    def _on_lineedit_edited(self):
+        """When the user hits enter or loses focus on the LineEdit widget."""
+        text = self.lineEdit.text()
+        rgb_match = rgba_regex.match(text)
+        if rgb_match:
+            text = [float(x) for x in rgb_match.groups() if x]
+        self.colorSwatch.setColor(text)
+
+    @Slot(np.ndarray)
+    def _on_swatch_changed(self, color: np.ndarray):
+        """Receive QColorSwatch change event, update the lineEdit, re-emit."""
+        self.lineEdit.setText(color)
+        self.color_changed.emit(color)
+
+
+class QColorSwatch(QFrame):
+    """A QFrame that displays a color.
+
+    The internal color representation (self._color) is always a 1x4 np.ndarray
+    """
+
+    color_changed = Signal(np.ndarray)
+
+    def __init__(
+        self,
+        parent: Optional[QWidget] = None,
+        tooltip: Optional[str] = None,
+        initial_color: Optional[ColorType] = None,
+    ):
+        """Create a new ColorSwatch
+
+        Parameters
+        ----------
+        parent : QWidget, optional
+            parent widget, by default None
+        tooltip : Optional[str], optional
+            Tooltip when hovering on swatch,
+            by default 'click to set color'
+        initial_color : ColorType, optional
+            initial color, by default will be transparent
+        """
+        super().__init__(parent)
+        self.setObjectName('colorSwatch')
+        self.setToolTip(tooltip or 'click to set color')
+        self.setCursor(Qt.PointingHandCursor)
+
+        self.color_changed.connect(self._update_swatch_style)
+        self._color: np.ndarray = TRANSPARENT
+        if initial_color is not None:
+            self.setColor(initial_color)
+
+    def color(self):
+        """Return the current color"""
+        return self._color
```

Comment:
Should this be implemented as a property?

Reply:
I'm always torn when doing Qt stuff, whether it's better to mimic the Qt API or the pythonic API.

Reply:
I'm inclined to go with Pythonic in ambiguous cases. =) Also, I think in cases where we are creating a class that we will *mostly* interact with using our own methods, rather than the built-in Qt methods, we should use a Pythonic API.